### PR TITLE
Publish images to Quay kroxylicious/kroxylicious rather than kroxylicious-developer

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

We decided in discussion in #597, that at least for the moment, we'd publish release images exclusively to https://quay.io/repository/kroxylicious/kroxylicious.   The release images already include filters such as EnvelopeEncryption, so they are suitable for someone who want to try out the filters.

We'll stop publishing to kroxylicious-developer.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by k-wall** [comment]
To help any existing users we should push the previous release images from https://quay.io/repository/kroxylicious/kroxylicious-developer?tab=tags to https://quay.io/repository/kroxylicious/kroxylicious and put a description onto https://quay.io/repository/kroxylicious/kroxylicious-developer saying it is not going to be used anymore.

**Comment by k-wall** [comment]
@strimzi-ci run system tests

**Comment by robobario** [comment]
Need to look at how this plays with system tests, I can see they write to `kroxylicious.io/kroxylicious` already and then delete the images afterwards https://quay.io/repository/kroxylicious/kroxylicious?tab=history. Maybe it should be reversed after this change, so kroxylicious-developer is used temporarily? Pushing test tags temporarily through the main artifact probably not what we want.

Also the system test code needs updating (in lockstep/after the change? @franvila) as it references `kroxylicious-developer` as the default string parameter and the image used when run against the main merge.

**Review by robobario — APPROVED** [review]
Changes look good, just need figure out which order to update systemtests

File: kubernetes-examples/envelope-encryption/base/kroxylicious/kroxylicious-deployment.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ spec:
     spec:
       containers:
       - name: kroxylicious
-        image: quay.io/kroxylicious/kroxylicious-developer:0.4.1
+        image: quay.io/kroxylicious/kroxylicious:0.5.0-SNAPSHOT
```
Comment (SamBarker) [inline kubernetes-examples/envelope-encryption/base/kroxylicious/kroxylicious-deployment.yaml:26]: hmm, we do want to get this back to a non snapshot version ASAP.

**Review by SamBarker — APPROVED** [review]
>Need to look at how this plays with system tests, I can see they write to kroxylicious.io/kroxylicious already and then delete the images afterwards https://quay.io/repository/kroxylicious/kroxylicious?tab=history. Maybe it should be reversed after this change, so kroxylicious-developer is used for these temporary artefacts? Pushing test tags temporarily through the main artifact probably not what we want.

@robobario is 100% right we shouldn't be shuffling testing builds through the production image repo! I'd argue if they need to be published at all they should go through an alternative org. `quay.io/kroxylicious-testing/kroxylicious-[testing|???]` replace the final testing with something appropriate...

**Comment by franvila** [comment]
> Need to look at how this plays with system tests, I can see they write to `kroxylicious.io/kroxylicious` already and then delete the images afterwards https://quay.io/repository/kroxylicious/kroxylicious?tab=history. Maybe it should be reversed after this change, so kroxylicious-developer is used for these temporary artefacts? Pushing test tags temporarily through the main artifact probably not what we want.
> 
> Also the system test code needs updating (in lockstep/after the change? @franvila) as it references `kroxylicious-developer` as the default string parameter and the image used when run against the main merge.

Since a recent change, the system tests for a PR are uploading images to https://quay.io/repository/kroxylicious/kroxylicious using an ephemeral version related to the commit sha and will be removed automatically after 8 hours. But as @SamBarker suggest, I would go with something like that, moving to another repo for testing.

**Comment by k-wall** [comment]
> Since a recent change, the system tests for a PR are uploading images to https://quay.io/repository/kroxylicious/kroxylicious using an ephemeral version related to the commit sha and will be removed automatically after 8 hours. But as @SamBarker suggest, I would go with something like that, moving to another repo for testing.

Yes, agreed.  I can make a change to the pipeline. https://quay.io/repository/kroxylicious/kroxylicious-pr okay?

File: kubernetes-examples/envelope-encryption/base/kroxylicious/kroxylicious-deployment.yaml (hunk 1 above)
Comment (k-wall) [reply kubernetes-examples/envelope-encryption/base/kroxylicious/kroxylicious-deployment.yaml:26]: I believe main should refer to main's snapshot.
the release mechanism updates the version number in these artefacts to main the release version.

**Comment by k-wall** [comment]
@strimzi-ci run system tests

**Pull request merged**